keanu-project/src/test/java/io/improbable/keanu/algorithms/mcmc/proposal/AcceptanceRateTrackerTest.java

```diff
@@ -0,0 +1,100 @@
+package io.improbable.keanu.algorithms.mcmc.proposal;
+
+import com.google.common.collect.ImmutableSet;
+import io.improbable.keanu.vertices.Vertex;
+import org.junit.Before;
+import org.junit.Test;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.equalTo;
+import static org.mockito.Mockito.mock;
+
+public class AcceptanceRateTrackerTest {
+
+    private Vertex vertex1 = mock(Vertex.class);
+    private Vertex vertex2 = mock(Vertex.class);
+    Proposal proposal;
+    private AcceptanceRateTracker acceptanceRateTracker;
+
+    @Before
+    public void createTracker() throws Exception {
+        acceptanceRateTracker = new AcceptanceRateTracker();
+    }
+
+    @Test
+    public void theAcceptanceRateForAnUnrecognisedSetOfVerticesIsNaN() {
+        assertThat(acceptanceRateTracker.getAcceptanceRate(ImmutableSet.of(vertex1)), equalTo(Double.NaN));
+    }
+
+    @Test
+    public void youCanTrackTheAcceptanceRateForASingleVertex() {
+        proposal = new Proposal();
+        proposal.addListener(acceptanceRateTracker);
+        proposal.setProposal(vertex1, 1.);
+        proposal.apply();
+        proposal.reject();
+        assertThat(acceptanceRateTracker.getAcceptanceRate(ImmutableSet.of(vertex1)), equalTo(0.));
+
+        proposal = new Proposal();
+        proposal.addListener(acceptanceRateTracker);
+        proposal.setProposal(vertex1, 2.);
+        proposal.apply();
+        assertThat(acceptanceRateTracker.getAcceptanceRate(ImmutableSet.of(vertex1)), equalTo(0.5));
```

Comment:
There has been three proposals, one has been accepted. Shouldn't this be equal to `1/3`?

Reply:
see my explanation above about "applied"